### PR TITLE
User hover card

### PR DESCRIPTION
**Pull request by jdalsem**
- added icons to all user hover menu items
- new layout for the user hover menu

afbeelding

and with expanded admin options

afbeelding

todo:
- [x] write upgrade notice about changes in these views
- [x] check responsiveness behaviour

### CODE REVIEW

**Comment by hypeJunction** [comment]
I like the card, but we need to remove all that junk from user icons.

**Comment by hypeJunction** [comment]
Can we make it a hover tooltip and trigger on hover or tabbed focus on mobile?

**Comment by mrclay** [comment]
> Can we make it a hover tooltip and trigger on hover or tabbed focus on mobile?

Can be tackled independently I'd think.

> we need to remove all that junk from user icons

Problem is who decides which menu items stay? It's easy enough to remove items by plugin.

**Comment by jdalsem** [comment]
> we need to remove all that junk from user icons
> 
> Problem is who decides which menu items stay? It's easy enough to remove items by plugin.

i think @hypeJunction means most of those user hover admin actions are user 'entity' menu items, and could go to a entity menu dropdown with actions...

This PR does not change any existing menu items (other than adding icons). Moving them somewhere else is fine by me.



> Can we make it a hover tooltip and trigger on hover or tabbed focus on mobile?
> 
> Can be tackled independently I'd think.

Was thinking the same. Can do a follow up PR for that. Also was thinking about a user/owner link page element (the user link in the by_line) that also shows the card on hover (like on FB)

**Comment by jdalsem** [comment]
Ready to be merged

**Comment by hypeJunction** [comment]
The choice of icons is rather strange, but ok. 
Can you reposition it to left-top corner of the icon so it overlays it?

**Comment by jdalsem** [comment]
> The choice of icons is rather strange, but ok.

which do you find strange?

**Review by jeabakker — APPROVED** [review]

**Comment by jdalsem** [comment]
> Can you reposition it to left-top corner of the icon so it overlays it?

done

**Comment by hypeJunction** [comment]
We can change them later I suppose. Can you show user's own popup?
I think this window will grow out of proportion on some sites. IMO, we should keep first three items of actions, put the rest into toggled part.

**Comment by jdalsem** [comment]
afbeelding



> I think this window will grow out of proportion on some sites.

theoretically this could be, but it scales very nicely



> IMO, we should keep first three items of actions, put the rest into toggled part.

There are not that many actions/menu items in the right column anyway...

**Comment by hypeJunction** [comment]
Ok, let it sit for a bit, I won't have to look at anything until next week.

**Comment by jdalsem** [comment]
no need to wait until next week to merge it right... don't want things to get stale

File: engine/lib/users.php
Comment (hypeJunction) [inline engine/lib/users.php]: Should check if self

File: mod/logbrowser/start.php
Comment (hypeJunction) [inline mod/logbrowser/start.php]: I would use history icon

**Comment by mrclay** [comment]
Choice of hourglass makes that link look like a search input.

File: mod/reportedcontent/start.php
Comment (hypeJunction) [inline mod/reportedcontent/start.php]: Let's not user Elgg icons anymore. Phase them out. I would push this to the very end of the list. Bullhorn perhaps?

File: views/default/elements/navigation.css.php
Comment (hypeJunction) [inline views/default/elements/navigation.css.php]: Vendor prefixes?

File: views/default/navigation/menu/user_hover.php
Comment (hypeJunction) [inline views/default/navigation/menu/user_hover.php]: We should support other sections

File: views/default/navigation/menu/user_hover.php
Comment (hypeJunction) [inline views/default/navigation/menu/user_hover.php]: Get rid of?

File: views/default/navigation/menu/user_hover/contents.php
Comment (hypeJunction) [inline views/default/navigation/menu/user_hover/contents.php]: Needed?

**Comment by mrclay** [comment]
Please hold. I can't get CSS to load...

**Comment by mrclay** [comment]
I'm not crazy that the icons vary in width causing the link texts to not align. Even more noticeable for links without icons (e.g. Login as, Add admin note). I'm guessing that's not easily addressed.

**Comment by hypeJunction** [comment]
I can fix that. Use set width with center align

File: engine/lib/users.php
Comment (mrclay) [inline engine/lib/users.php]: 👍  I've used this for my own test admin user before, but probably rarely needed in this menu.

File: views/default/elements/navigation.css.php
Comment (mrclay) [inline views/default/elements/navigation.css.php]: csscrush will handle this, but I agree this PR should stand alone as the codebase is today.

File: views/default/navigation/menu/user_hover.php
Comment (mrclay) [inline views/default/navigation/menu/user_hover.php]: We can extend via a follow-up PR.

**Review by mrclay — APPROVED** [review]
Most gripes about this are legacy stuff/fixable in separate PRs.

**Pull request merged**
